### PR TITLE
add composer.json for simply grabbing bibtexbrowser with composer

### PR DESCRIPTION
**Pull request by paxperscientiam**
Hello @monperrus 

I have introduced a simple `composer.json` file that allows composer users to grab `monperrus/bibtexbrowser` even though it is not registered with packagist.

The downside of not having registered with packagist -- perhaps you could do this? -- is that end-user composer.json is a tad more complicated.

For the composer novice, they should know that they have to specify the git endpoint as a repository. In order for the end-user to "require" the master branch, it must be specified as "dev-master" in the end-user's composer.json file. Likewise, if you choose to split this off into, say, "composer-branch", then the end-user would need to require (that's composer parlance) "dev-composer-branch."

EG:
````json
{
    "repositories": [
	{
	    "type": "vcs",
	    "url": "https://github.com/monperrus/bibtexbrowser.git"
	}
    ],
    "require": {
        "monperrus/bibtexbrowser": "dev-master"
    }
}
````

What do you think?

Incorporating autoloading is perfectly doable, but it would require some restructuring of your code. Basically, the code would need to be wrapped in a class (probably over simplifying).

EDIT: To be clear, I'd be happy to get auto-loading working if you're interested. However, you'd probably want this to be it's own release as it would be incompatible with how people are currently using bibtexbrowser.

### CODE REVIEW

**Comment by monperrus** [comment]
That's great. Can you add Question&Answer explaining how to use bibtexbrowser with composer in the documentation file (`bibtexbrowser-documentation.wiki`)

> The downside of not having registered with packagist -- perhaps you could do this?

Of course. How to do this?

**Comment by paxperscientiam** [comment]
I believe you just head on over to [packagist.org](https://packagist.org/). Not sure if the change was recent, but you can now login using your github credentials for quick pairing.

From there...
* click on submit at the top, which will take you to a page where you submit the link to, in this case, bibtexbrowser
* [Here](https://packagist.org/about#how-to-update-packages) they describe how to set up a hook between Github and packagist so that latest changes are reflected on packagist after push. Without this, I think the repo is only crawled once per week.

Once you do this, I can run tests on my end to make sure composer is grabbing the library correctly. Subsequently, you might consider a separate release with autoloading...but that's a story for another day :)

**Comment by paxperscientiam** [comment]
@monperrus OK, I updated wiki. Good?

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "monperrus/bibtexbrowser",
+    "description": "bibtexbrowser is a PHP script that creates publication lists from Bibtex files. bibtexbrowser is stable, mature and easy to install.",
+    "homepage": "http://www.monperrus.net/martin/bibtexbrowser/",
+    "license": "GPL-2.0+",
+    "authors": [
+	{
+	    "name": "Martin Monperrus",
+	    "homepage": "http://www.monperrus.net",
```
Comment (monperrus) [inline composer.json]: http://www.monperrus.net/martin/

File: bibtexbrowser-documentation.wiki [hunk 2]
```diff
@@ -513,6 +514,24 @@ Use ''bibtexbrowser-cli.php''. For instance:
 php bibtexbrowser-cli.php mybib.bib --id classical --set-title \"a new title\" --id with_abstract --set-title \"a new title\" --set-year 1990
 </pre>
 
+====How to use with composer?====
+Bibtexbrowser can now be downloaded with composer, though not yet with packagist.
```
Comment (monperrus) [inline bibtexbrowser-documentation.wiki]: Bibtexbrowser can now be downloaded with composer and packagist

(I will create the packagist just after having merged this one)

Comment (paxperscientiam) [inline bibtexbrowser-documentation.wiki]: In which case, I need to update the project's composer.json to reflect this!

Comment (paxperscientiam) [inline bibtexbrowser-documentation.wiki]: Strike that last remark, I was thinking of the end-user. I will update the wiki tho.

File: composer.json (hunk 1 above)
Comment (paxperscientiam) [inline composer.json]: @monperrus OK, updated the homepage entry!

**Comment by paxperscientiam** [comment]
@monperrus Just to be sure, could you clarify which versions of PHP bibtexbrowser is compatible with? I don't recall where I read it, but I thought PHP-4.X.X to PHP-5.X.X.

Thanks.

**Comment by monperrus** [comment]
should be all of them incl. 7., although Travis only tests 5 and 7.

**Comment by monperrus** [comment]
thanks a lot!

**Comment by monperrus** [comment]
now on packagist: https://packagist.org/packages/monperrus/bibtexbrowser

**Pull request merged**